distributed/worker_state_machine.py

```diff
@@ -40,7 +40,7 @@
 from distributed.sizeof import safe_sizeof as sizeof
 from distributed.utils import recursive_to_dict
 
-logger = logging.getLogger(__name__)
+logger = logging.getLogger("distributed.worker.state_machine")
```

Comment:
I think with this change, we'd see the logs now twice when accessing them with `Worker.get_logs()`, i.e. the DequeHandler is registered twice.

We only need to attach the DequeHandler to the top-level logger, i.e. `distributed.worker` and the propagation will bubble it up the hierarchy,

In fact, we could consider attaching the deque handler to the root logger or at least `distributed`. This way, we wouldn't ever miss any distributed logs anymore, regardless of how the loggers are named

Reply:
relevant code line is https://github.com/dask/distributed/blob/69b74f9f0aa46dac7facb960f1b0aaac69121e5e/distributed/worker.py#L575 where we're registering two loggers. Once should be enough

Reply:
I changed worker.py:575 for this exact reason.

Reply:
Also see new test in test_client.py which tests that we're getting everything once.

Reply:
sorry, I missed this.